x/bank/client/cli/encode.go

```diff
@@ -0,0 +1,76 @@
+package cli
+
+import (
+	"encoding/base64"
+	"io/ioutil"
+	"os"
+
+	"github.com/spf13/cobra"
+	amino "github.com/tendermint/go-amino"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+)
+
+type encodeReq struct {
+	Tx auth.StdTx `json:"tx"`
+}
+
+// PrintOutput requires a Stringer, so we wrap string
+type encodeResp string
+
+func (e encodeResp) String() string {
+	return string(e)
+}
+
+// GetEncodeCommand returns the encode command to take a JSONified transaction and turn it into
+// Amino-serialized bytes
+func GetEncodeCommand(codec *amino.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "encode [file]",
+		Short: "encode transactions generated offline",
+		Long: `Encode transactions created with the --generate-only flag and signed with the sign command.
+Read a transaction from <file>, serialize it to the Amino wire protocol, and output it as base64.
+If you supply a dash (-) argument in place of an input filename, the command reads from standard input.`,
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			cliCtx := context.NewCLIContext().WithCodec(codec)
+			req, err := readAndUnmarshalRequest(cliCtx.Codec, args[0])
+			if err != nil {
+				return err
+			}
+
+			txBytes, err := cliCtx.Codec.MarshalBinaryLengthPrefixed(req.Tx)
+			if err != nil {
+				return err
+			}
+
+			// Encode the bytes to base64
+			txBytesBase64 := base64.StdEncoding.EncodeToString(txBytes)
+
+			// Write it back
+			response := encodeResp(txBytesBase64)
+			cliCtx.PrintOutput(response)
+			return nil
+		},
+	}
+
+	return client.PostCommands(cmd)[0]
+}
+
+func readAndUnmarshalRequest(cdc *amino.Codec, filename string) (req encodeReq, err error) {
```

Comment:
do we have this functionality anywhere else @alessio?

Reply:
Good catch @alexanderbez! Indeeed we do - from `x/bank/client/cli/broadcast`:

```golang
func readAndUnmarshalStdTx(cdc *amino.Codec, filename string) (stdTx auth.StdTx, err error) {
	var bytes []byte
	if filename == "-" {
		bytes, err = ioutil.ReadAll(os.Stdin)
	} else {
		bytes, err = ioutil.ReadFile(filename)
	}
	if err != nil {
		return
	}
	if err = cdc.UnmarshalJSON(bytes, &stdTx); err != nil {
		return
	}
	return
}
```

@ducembarr I would:

1. get rid of this one
2. move `readAndUnmarshalStdTx` from his location in the aforementioned package into `client/utils.go`
3. rename it: `readAndUnmarshalStdTx` -> `ReadStdTxFromFile`

If you don't spare enough time, I can do it in a separate PR

Reply:
@alessio I'm happy to do it, but which `client/utils.go` are you referencing?  `client/utils/utils.go` ?  Or a new `x/bank/client/utils.go` ?

Reply:
The former please

Reply:
Hang on, please drop it in `x/auth/client`. Feel free to create a `x/auth/client/util.go` file - a test case would be appreciated too :)

Reply:
Done; thanks.